File: src/main/java/com/heekng/celloct/repository/ManagerRepository.java
package com.heekng.celloct.repository;

import com.heekng.celloct.entity.Manager;
import org.springframework.data.jpa.repository.EntityGraph;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.Optional;

public interface ManagerRepository extends JpaRepository<Manager, Long> {

    Optional<Manager> findByMemberIdAndShopId(Long memberId, Long shopId);

    List<Manager> findByMemberId(Long memberId);

    List<Manager> findListByShopId(Long shopId);

    @EntityGraph(attributePaths = {"member"})
    Manager findWithMemberById(Long id);
}

File: src/main/kotlin/com/heekng/celloct/repository/ManagerRepository.kt
package com.heekng.celloct.repository

import com.heekng.celloct.entity.Manager
import org.springframework.data.jpa.repository.EntityGraph
import org.springframework.data.jpa.repository.JpaRepository

interface ManagerRepository : JpaRepository<Manager, Long>, ManagerRepositoryCustom {

    @EntityGraph(attributePaths = ["shop"])
    fun findWithShopByMemberId(memberId: Long): List<Manager>

    @EntityGraph(attributePaths = ["member"])
    fun findWithMemberById(id: Long): Manager

    fun findByShopIdAndId(shopId: Long, id: Long): Manager?
}